Extension/src/LanguageServer/client.ts

```diff
@@ -1048,17 +1049,39 @@ export class DefaultClient implements Client {
             }
             if (index === paths.length - 2) {
                 action = "help";
+                // Because we need to conditionally enable/disable steps to alter their contents,
+                // we need to determine which step is actually visible. If the steps change, this
+                // logic will need to change to reflect them.
+                let step: string = "ms-vscode.cpptools#";
+                if (!scanForCompilersDone) {
+                    step = step + "awaiting.activation.";
+                } else if (compilerDefaults?.knownCompilers === undefined || !compilerDefaults.knownCompilers.length) {
+                    step = step + "no.compilers.found.";
+                } else {
+                    step = step + "verify.compiler.";
+                }
                 switch (os.platform()) {
                     case 'win32':
-                        void vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217614");
-                        return;
+                        step = step + "windows";
+                        break;
                     case 'darwin':
-                        void vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217706");
-                        return;
+                        step = step + "mac";
+                        break;
                     default: // Linux
-                        void vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217615");
-                        return;
+                        step = step + "linux";
+                        break;
                 }
+                void vscode.commands.executeCommand(
```

Comment:
It seems like this should be `return` instead of `void`, right?

Reply:
Should it?  Seems like we don't need to pause our own execution waiting for VS Code to process this command.

Reply:
I don't see how `return` pauses our execution. Wouldn't that be `await`?

Reply:
Oh, you mean if VS Code itself awaited the result? Okay, sure, `void` makes sense then.

Reply:
Or should we use `return` in case the caller actually wants to wait for the command to execute?

Reply:
What I think is confusing is that some paths in this function return a promise and others don't. Prior to #11085, none of these awaitable functions were awaited nor were their promises returned.  I reverted a change that was made before #11085.